### ISSUE TITLE
QAM: If you're not logged in, the link from the text goes to the view by school, not the reminders screen

### ISSUE THREAD

**Issue by michiQ** [yadaguru/yadaguru-app#154]

**Comment by michiQ** [comment]
So we've got an awesome screen as the landing page from the text message. But if we aren't logged into the app, we go to the login screen and then, when you log in, you go to the view by schools page, not the text message landing page. Wondering if we should keep users logged in to work arouned this?

**Comment by michiQ** [comment]
![img_1273](https://cloud.githubusercontent.com/assets/8203279/23518970/3e8163f2-ff43-11e6-93a2-ba1f9ba9f260.PNG)
This is that nice landing page we want them to see

**Comment by bobderrico80** [comment]
Yes, we will have to track the URL the user came in from when they are redirected to the login page.

Most of the time, the users won't need to login again, since in most cases, they will probably be using YadaGuru on the same device that happens to be associated with the phone number.

Currently, the access tokens are not set to expire, so unless they explicitly logout or are using a different device, they should go to the screen. That said, the logic to handle the exceptional case should be added.

**Comment by michiQ** [comment]
ok - going to close this.  I think the problem is that I'm using two phones, two numbers, and  my comp.

**Comment by michiQ** [comment]
I'm still thinking this one is an issue. I've got yadaguru open on both of my devices, logged in.  I got my link in my text today but when I clicked the text link I still got the login screen.  I would actually like someone else to test this since I'm worried about my multiple device issue.  I just want to rule out that it *is* the device issue and that this will work normally on one device.

**Comment by bobderrico80** [comment]
This _should_ work normally if the user is logged in and getting messages on the same device. In any case, we should still be redirecting them to the correct screen afterwards, so I am all for reopening it.

**Comment by michiQ** [comment]
yeah - I think it might be me but just wanted to see if someone else could test.  I don't want to use my qam number 3014420047 because it's all over the place due to testing.  So was hoping maybe you could give it a whirl in qa.  Or if you want to add my husband to qam I can test with him - his device is clean 240-271-1828

**Comment by bobderrico80** [comment]
This behavior (while incorrect) is consistent with the way things are coded. When you click the link, and you are not logged in you will be redirected from the link to the login page.

The login page's behavior is set to redirect you to schools after you login. What we need to do is change the logic so that when a user is redirected to login, we store where they came from (in this case it would be the individual day reminder page), and then redirect them to that location, rather than automatically redirecting them to schools, as is what currently happen.

I can test on QA, but I am 99% sure this is what is happening.

**Comment by michiQ** [comment]
Ah I see.  That makes sense.
